firmware/defmt-test/macros/src/lib.rs

```diff
@@ -61,10 +61,8 @@ fn tests_impl(args: TokenStream, input: TokenStream) -> parse::Result<TokenStrea
                 let attr = match test_kind {
                     Some(it) => it,
                     None => {
-                        return Err(parse::Error::new(
-                            f.span(),
-                            "function requires `#[init]` or `#[test]` attribute",
-                        ));
+                        untouched_tokens.push(Box::new(f));
+                        continue;
```

Comment:
This might paint us into a corner, since it means we can't add other attributes in the future

Reply:
I thought about this case as well. 
Keeping the attribute requirement for functions, but allowing everything else is ok?

With the other code change, it would mean, that we can't add any attribute to other items in the future as well, right?

Reply:
We can't add a newer attribute in a backwards-compatible way under the assumption that every new attribute(-name) is already used / introduced by another crate. 

Do I understand that correctly? 

What about namespacing? Every attribute from defmt-test lives under `defmt_test::init` or `defmt_test::test` and we generate a `use defmt_test::test` implicitly. Would this be possible or even useful?

Reply:
> We can't add a newer attribute in a backwards-compatible way under the assumption that every new attribute(-name) is already used / introduced by another crate.

Yeah that's correct. It wouldn't be too hard to avoid builtin attributes, but you can use arbitrary procedural macros too, which might define their own stuff.

> Every attribute from defmt-test lives under `defmt_test::init` or `defmt_test::test` and we generate a `use defmt_test::test` implicitly.

There's no name resolution inside the test module, so we can't insert a `use`, but namespacing does sound like a possible solution for attributes we add later (unfortunately requiring the user to write out a rather long path).